modules/core/src/v2/coins/algo.ts

```diff
@@ -352,27 +414,25 @@ export class Algo extends BaseCoin {
       throw new Error(`prv must be a string, got type ${typeof prv}`);
     }
 
-    if (
-      !_.has(params.txPrebuild, 'keys[0]') ||
-      !_.has(params.txPrebuild, 'keys[1]') ||
-      !_.has(params.txPrebuild, 'keys[2]')
-    ) {
+    if (!_.has(params.txPrebuild, 'keys')) {
       throw new Error('missing public keys parameter to sign transaction');
     }
 
     if (!_.isNumber(addressVersion)) {
       throw new Error('missing addressVersion parameter to sign transaction');
     }
 
-    // we need to re-encode our public keys using algosdk's format
-    const keys = [params.txPrebuild.keys[0], params.txPrebuild.keys[1], params.txPrebuild.keys[2]];
-
-    // re-encode sk from our prv (this acts as a seed out of the keychain)
-    const seed = Seed.decode(prv).seed;
-    const pair = generateAccountFromSeed(seed);
-    const sk = pair.sk;
-
-    return { txHex, addressVersion, keys, sk, isHalfSigned };
+    const signers = params.txPrebuild.keys.map((key) => {
+      // if we are receiving addresses do not try to convert them
+      if (!accountLib.Algo.algoUtils.isValidAddress(key)) {
+        return accountLib.Algo.algoUtils.publicKeyToAlgoAddress(accountLib.Algo.algoUtils.toUint8Array(key));
+      }
+      return key;
+    });
+    // TODO(https://bitgoinc.atlassian.net/browse/STLX-6067): fix the number of signers using
+    // should be similar to other coins implementation
+    const numberSigners = Math.trunc(signers.length / 2) + 1;
```

Comment:
I'm quite confused here, does each signer appear in the `signers` list twice? Also the JIRA has no additional info here. Could you please leave a comment here to explain the math?

Reply:
No, this is related to the number of signers required for the transaction to be valid. More than 50% of the signers are required to do that.

Reply:
Got it, makes sense. The variable name or a comment could be used to make that a bit more clear.